### PR TITLE
test(FIR-12467): improve unit test coverage

### PR DESCRIPTION
**Pull request by ptiurin**
### Description

Adding more tests to verify edge cases.

### CODE REVIEW

File: tests/unit/test_firebolt_adapter.py [hunk 1]
```diff
@@ -32,6 +35,16 @@ def connection():
     return connection
 
 
+class MockFireboltAdapter(FireboltAdapter):
```
Comment (stepansergeevitch) [inline tests/unit/test_firebolt_adapter.py:38]: Do we use this one?

**Review by stepansergeevitch — APPROVED** [review]
LGTM, one comment

File: tests/unit/test_firebolt_adapter.py (hunk 1 above)
Comment (ptiurin) [reply tests/unit/test_firebolt_adapter.py:38]: You're right, that's some dead code
